Mathlib/Data/Bool/Basic.lean

```diff
@@ -81,8 +81,14 @@ theorem eq_iff_eq_true_iff {a b : Bool} : a = b ↔ ((a = true) ↔ (b = true))
   cases a <;> cases b <;> simp
 
 -- Porting note: new theorem
-theorem beq_eq_decide_eq {α} [DecidableEq α]
-    (a b : α) : (a == b) = decide (a = b) := rfl
+/- Even though `DecidableEq α` implies an instance of (`Lawful`)`BEq α`, we keep the seemingly
+    redundant typeclass assumptions so that the theorem is also applicable for types that have
+    overridden this default instance of `LawfulBeq α` -/
```

Comment:
```suggestion
redundant typeclass assumptions so that the theorem is also applicable for types that have
overridden this default instance of `LawfulBEq α` -/
```